File: product-accounting-backend/src/main/java/com/ichzerowan/accounting/model/coupon/Coupon.java
package com.ichzerowan.accounting.model.coupon;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Coupon {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    private String code;

    private double amount;

    private int count;

    private boolean archived;

    public Coupon() { }

    public Coupon(String code, double amount, int count) {
        this.code = code;
        this.amount = amount;
        this.count = count;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public boolean isArchived() {
        return archived;
    }

    public void setArchived(boolean archived) {
        this.archived = archived;
    }
}

File: product-accounting-frontend/product-accounting-frontend/models/Coupon.cs
using Newtonsoft.Json;

namespace product_accounting_frontend.models
{
    public class Coupon : EntityBase
    {
        [JsonProperty("code")]
        public string code { get; set; }
        [JsonProperty("amount")]
        public double amount { get; set; }
        [JsonProperty("count")]
        public int count { get; set; }
        [JsonProperty("archived")]
        public bool archived { get; set; }

        public Coupon()
        {

        }

        public Coupon(int id, string code, double amount, int count, bool archived)
        {
            this.id = id;
            this.code = code;   
            this.amount = amount;
            this.count = count;
            this.archived = archived;
        }
    }
}
